src/AddOn.Episerver.Settings/Core/SettingsService.cs

```diff
@@ -281,6 +325,37 @@ public void UpdateSettings(IContent content)
             }
         }
 
+        /// <summary>
+        /// Creates a settings folder for a site and adds it to the settings lookup
+        /// </summary>
+        public void CreateSiteSettingsRoot(SiteDefinition siteDefinition)
```

Comment:
Perhaps rename to ValidateOrCreateSettingsRoot or similar to indicate that this method does not always result in creating a root folder.